### PR TITLE
Implement automatic resource resolution

### PR DESCRIPTION
**Pull request by kosovo0275**

### CODE REVIEW

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/BaksmaliConfig.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    //    Theyre then added to the list of system resources, and are subsequently used
+    //    to annotate the Smali output everytime a resource is referenced
+    //    during the disassembly of the dex files, and will look something like:
+    //
+    //        # base.string.app_name
+    //        # base.layout.activity_main
+    public void loadResourceIds(ResTable resTable, File outDir) throws SAXException, IOException, AndrolibException {
+        this.loadFrameworkResources(resTable);
+        Map<String, File> resourceFiles = new HashMap<>();
+        resourceFiles.put(outDir.getName(), new File(outDir, "res/values/public.xml"));
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/BaksmaliConfig.java:73]: Can you explain this part? I'm not following why we'd generate our own `res/values/public.xml`. Wouldn't that conflict with the natural generation of it? Is that solely done to keep the API happy with baksmali that needs it?

It seems (albeit maybe memory heavy) - we can keep the resources in memory during disassembly and just reference them during this override of `loadResourceIds` with some dummy null file that actually queries that hashmap

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/BaksmaliConfig.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            for (ResResSpec spec : pkg.listResSpecs()) {
+                String resourceId = String.format("0x%08x", spec.getId().id);
+                String qualifiedResourceName = String.format("Android.%s.%s", spec.getType().getName(), spec.getName());
+                resourceIds.put(Integer.decode(resourceId), qualifiedResourceName);
+            }
+        } catch (AndrolibException ex) {
+            throw new AndrolibException("Could not parse framework resources", ex);
+        }
+    }
+
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/BaksmaliConfig.java:91]: minor nit - can we kill newline?

**Review by iBotPeaches — COMMENTED** [review]
This does look a lot smaller than the last attempt. The override of the method on baksmali seems much more elegant than last time.

One thing I'm curious about

**Review by abdelhak-zaaim — APPROVED** [review]

**Comment by brunoaduarte** [comment]
@kosovo0275 
```
dpi@ubuntu:~$ java -jar apktool-cli.jar b app_arm64-v8a app_arm64-v8a_mod.apk
I: Using Apktool 2.10.0-dirty with 20 thread(s).
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes3 folder into classes3.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes2 folder into classes2.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes4 folder into classes4.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes5 folder into classes5.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes6 folder into classes6.dex...
I: Checking whether resources has changed...
I: Building resources...
W: /home/dpi/app_arm64-v8a/res/layout-land/APKTOOL_DUMMYVAL_0x7f0e004c.xml:7: error: not well-formed (invalid token).
W: /home/dpi/app_arm64-v8a/res/layout-land/APKTOOL_DUMMYVAL_0x7f0e004c.xml: error: file failed to compile.
W: /home/dpi/app_arm64-v8a/res/layout/APKTOOL_DUMMYVAL_0x7f0e004b.xml:22: error: not well-formed (invalid token).
W: /home/dpi/app_arm64-v8a/res/layout/APKTOOL_DUMMYVAL_0x7f0e004b.xml: error: file failed to compile.
W: /home/dpi/app_arm64-v8a/res/layout/APKTOOL_DUMMYVAL_0x7f0e004c.xml:7: error: not well-formed (invalid token).
W: /home/dpi/app_arm64-v8a/res/layout/APKTOOL_DUMMYVAL_0x7f0e004c.xml: error: file failed to compile.
W: /home/dpi/app_arm64-v8a/res/layout/APKTOOL_DUMMYVAL_0x7f0e0b70.xml:2: error: not well-formed (invalid token).
W: /home/dpi/app_arm64-v8a/res/layout/APKTOOL_DUMMYVAL_0x7f0e0b70.xml: error: file failed to compile.
brut.androlib.exceptions.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/tmp/brut_util_Jar_21395298060234331251324167065958333960.tmp, compile, --dir, /home/dpi/app_arm64-v8a/res, --legacy, -o, /home/dpi/app_arm64-v8a/build/resources.zip]
shouldn't happen: null
shouldn't happen: null
shouldn't happen: null
shouldn't happen: null
shouldn't happen: null
shouldn't happen: null
app_arm64-v8a/smali_classes5/X/8uL.smali[1,0] required (...)+ loop did not match anything at input ''
app_arm64-v8a/smali_classes4/X/4fn.smali[1,0] required (...)+ loop did not match anything at input ''
app_arm64-v8a/smali_classes3/X/3YQ.smali[1,0] required (...)+ loop did not match anything at input ''
```
Any ideas how to bypass this obfuscation ? Thanks

**Comment by iBotPeaches** [comment]
We spoke in email about a different method that doesn't require BaksmaliOption tampering.

**Pull request closed without merging**